finite_difference/src/field_mod.f90

```diff
@@ -368,6 +373,17 @@ function r2d_field_constructor(grid,    &
     else
        self%data(:,:) = 0.0_go_wp
     end if
+
+    if (present(init_global_data)) then
+        dx = grid%subdomain%global%xstart - self%internal%xstart
+        dy = grid%subdomain%global%ystart - self%internal%ystart
+
```

Comment:
This *could* be parallelised in the same way as the loop at L365 but since this is just initialisation it could be left for now, especially since the zeroing does first touch. Perhaps just a TODO here (no need for an issue number).

Reply:
Done - I mean I added omp :)